src/solvers/dg_tree/dg_2d.jl

```diff
@@ -200,15 +200,15 @@ end
 
 # Calculate 2D twopoint flux (element version)
 @inline function calcflux_twopoint!(f1, f2, u::AbstractArray{<:Any,4}, element,
-                                    volume_flux, equations, dg::DG, cache)
+                                    mesh::TreeMesh{2}, equations, volume_flux, dg::DG, cache)
 
   for j in eachnode(dg), i in eachnode(dg)
-    # Set diagonal entries (= regular volume fluxes due to consistency)
+    # Pull the solution values at the node i,j
     u_node = get_node_vars(u, equations, dg, i, j, element)
-    flux1 = flux(u_node, 1, equations)
-    flux2 = flux(u_node, 2, equations)
-    set_node_vars!(f1, flux1, equations, dg, i, i, j)
-    set_node_vars!(f2, flux2, equations, dg, j, i, j)
+    # diagonal (consistent) part not needed since diagonal of
+    # dg.basis.derivative_split_transpose is zero!
+    set_node_vars!(f1, zero(u_node), equations, dg, i, i, j)
+    set_node_vars!(f2, zero(u_node), equations, dg, j, i, j)
```

Comment:
These are set to zero because the get overwritten later anyways, do I understand this correctly?

Reply:
They are multiplied by the diagonal elements of `derivative_split_transpose`, which are supposed to be zero (but may be of the order of `eps()`). Thus, we don't need to compute the `flux`es as we did before but we must ensure that there isn't some `NaN` or something similar stored accidentally. These parts are updated later in `calcflux_twopoint_nonconservative!`.

Reply:
Thanks!

Reply:
yes, they are set to zero for "safety" because the diagonal entries of the matrix `d_split_transpose` are also zero. We could potentially not do this `set_node_vars` to 0.0 if there is a performance hit